### PR TITLE
NN-3279 use offender search to determine if the prisoner has a life sentence term

### PR DESCRIPTION
**Pull request by yasinmustafa**
The original implementation was relying on a prison-api call to determine if the offender had a Life term, however that didnt quite work in all scenarios. It needed some extra derivation on the sentence terms - after a slack chat with other dps devs it turns out the prisoner search service does this derivation https://mojdt.slack.com/archives/C012FNNQ4A3/p1623854264294400

So changed it to call the elastic  prisoner-search-service which returns a flag which fits our purpose.

### CODE REVIEW

File: backend/controllers/prisonerProfile/prisonerQuickLook.js [hunk 1]
```diff
@@ -132,6 +125,18 @@ module.exports = ({ prisonerProfileService, prisonApi, telemetry }) => async (re
 
   trackEvent(telemetry, username, activeCaseLoad)
 
+  const extractLifeImprisonmentStatus = async () => {
+    const [prisonerDetailsResponse] = await Promise.all(
```
Comment (yasinmustafa) [inline backend/controllers/prisonerProfile/prisonerQuickLook.js:129]: I kept this call separate from the other api calls (calls the elstic prisoner search service).. as it is only used to determine if the offender is on a life sentence (at the moment). Therefore only runs in the specific scenario when it's needed.

Although, it does return a lot of the data already on the quicklook tab and is faster than calling the prison-api. So we could refactor to use this call to populate most (maybe all) of the data

File: backend/controllers/prisonerProfile/prisonerQuickLook.js [hunk 2]
```diff
@@ -132,6 +125,18 @@ module.exports = ({ prisonerProfileService, prisonApi, telemetry }) => async (re
 
   trackEvent(telemetry, username, activeCaseLoad)
 
+  const extractLifeImprisonmentStatus = async () => {
```
Comment (sp-watson) [inline backend/controllers/prisonerProfile/prisonerQuickLook.js:128]: You don't need to use Promise.all here

File: backend/controllers/prisonerProfile/prisonerQuickLook.js [hunk 3]
```diff
@@ -158,7 +163,7 @@ module.exports = ({ prisonerProfileService, prisonApi, telemetry }) => async (re
               sentenceData.sentenceDetail &&
               sentenceData.sentenceDetail.releaseDate &&
               moment(sentenceData.sentenceDetail.releaseDate).format('D MMMM YYYY')) ||
-            extractLifeImprisonmentStatus(prisonerDataResponse, prisoner, unableToShowDetailMessage),
+            (await extractLifeImprisonmentStatus()),
```
Comment (sp-watson) [inline backend/controllers/prisonerProfile/prisonerQuickLook.js:166]: This function name is a bit misleading - actually it gets the Imp Status label

**Review by sp-watson — APPROVED** [review]

File: backend/controllers/prisonerProfile/prisonerQuickLook.js (hunk 2 above)
Comment (yasinmustafa) [reply backend/controllers/prisonerProfile/prisonerQuickLook.js]: Thanks Steve - good spot, done

File: backend/controllers/prisonerProfile/prisonerQuickLook.js (hunk 3 above)
Comment (yasinmustafa) [reply backend/controllers/prisonerProfile/prisonerQuickLook.js]: Agreed - renamed

**Review by sp-watson — APPROVED** [review]

**Pull request merged**